public/app/features/dashboard-scene/scene/DashboardDatasourceBehaviour.tsx

```diff
@@ -38,15 +49,28 @@ export class DashboardDatasourceBehaviour extends SceneObjectBase<DashboardDatas
       return;
     }
 
+    // find the source panel referenced in the the dashboard ds query
     const panelId = dashboardQuery.panelId;
     const vizKey = getVizPanelKeyForPanelId(panelId);
-    const panel = findVizPanelByKey(dashboard, vizKey);
+    const sourcePanel = findVizPanelByKey(dashboard, vizKey);
+
+    if (!(sourcePanel instanceof VizPanel)) {
+      return;
+    }
 
-    if (!(panel instanceof VizPanel)) {
+    //check if the source panel is a library panel and wait for it to load
+    if (isLibraryPanel(sourcePanel)) {
+      const libraryPanel = getLibraryPanelBehavior(sourcePanel);
+      if (!libraryPanel) {
+        throw new Error('Could not find LibraryPanelBehavior for panel');
+      }
+      libraryPanelSub = libraryPanel.subscribeToState((newLibPanel, oldLibPanel) => {
+        this.handleLibPanelStateUpdates(newLibPanel, oldLibPanel, queryRunner, sourcePanel);
+      });
```

Comment:
```suggestion
const libPanelBehavior = getLibraryPanelBehavior(sourcePanel);
if (libPanelBehavior) {
    libraryPanelSub = libPanelBehavior.subscribeToState((newLibPanel, oldLibPanel) => {
    this.handleLibPanelStateUpdates(newLibPanel, oldLibPanel, queryRunner, sourcePanel);
});
```

just a shorter form, I am not sure why your passing the queryRunner to handleLibPanelStateUpdates, it's not going to be the correct queryRunner (you can only get this after lib panel has loaded) 

And should you not check for `!libPanelBehavior.state.isLoaded`. ?

Reply:
the passed queryRunner is not the one from the lib panel, but from the dashboardDS panel, so I think it should be alright.